src/order/boolean_algebra.lean

```diff
@@ -19,10 +41,12 @@ export has_compl (compl)
 
 postfix `ᶜ`:(max+1) := compl
 
-/-- A boolean algebra is a bounded distributive lattice with a
-  complementation operation `-` such that `x ⊓ - x = ⊥` and `x ⊔ - x = ⊤`.
-  This is a generalization of (classical) logic of propositions, or
-  the powerset lattice. -/
+/-- A Boolean algebra is a bounded distributive lattice with:
+1. A complement operator `ᶜ` such that `x ⊓ xᶜ = ⊥` and `x ⊔ xᶜ = ⊤`, and
+2. A set difference operation `\` such that `x \ y = x ⊓ yᶜ`.
```

Comment:
I'm not sure I like this. A quick read could result in thinking that the set difference operation is new structure, rather than merely notation for existing structure.

Reply:
I'm probably missing something, but both `compl` and `sdiff` are new structure on the lattice, right? Since `boolean_algebra` extends both `has_compl` and `has_sdiff`.

Reply:
Granted, the fact that `sdiff` is bundled with `boolean_algebra` is not essential and might be confusing to people who are more famliar with the traditional definition. Not sure how best to describe what's going on here though.

Reply:
How about something like:
```suggestion
/-- A Boolean algebra is a bounded distributive lattice with:
a complement operator `ᶜ` such that `x ⊓ xᶜ = ⊥` and `x ⊔ xᶜ = ⊤`.
For convenience, it must also provide a set difference operation `\` defined
as `x \ y = x ⊓ yᶜ`.
```

Reply:
That looks good to me, modulo some tweaks:
```suggestion
/-- A Boolean algebra is a bounded distributive lattice with
a complement operator `ᶜ` such that `x ⊓ xᶜ = ⊥` and `x ⊔ xᶜ = ⊤`.
For convenience, it must also provide a set difference operation `\`
satisfying `x \ y = x ⊓ yᶜ`.
```

Reply:
I had a quick go at using `opt_param` to provide the set difference automatically, but I think there's something screwy with accessing old_structure_cmd_fields so it didn't work.

Reply:
Thanks, I wondered if something like that could work.